src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1374 @@
+
+    // Create a new file; the file and the class within will have
+    // "Minimized" postpended.
+    String minimizedFileName =
+        new StringBuilder(filePath).insert(filePath.lastIndexOf('.'), SUFFIX).toString();
+    File minimizedFile = new File(minimizedFileName);
+    // Rename the overall class to [original class name][suffix].
+    String origClassName = FilenameUtils.removeExtension(originalFile.getName());
+    new ClassRenamer().visit(compUnit, new String[] {origClassName, SUFFIX});
+    // Write the compilation unit to the minimized file.
+    writeToFile(compUnit, minimizedFile);
+
+    // Compile the Java file and check that the exit value is 0.
+    if (compileJavaFile(minimizedFile, classPath, packageName, timeoutLimit) != 0) {
+      System.err.println("Error when compiling file " + filePath + ". Aborting.");
+      return false;
+    }
+
+    String runResult = runJavaFile(minimizedFile, classPath, packageName, timeoutLimit);
+
+    // expectedOutput is a map from method name to failure stack trace
+    // with line numbers removed.
+    Map<String, String> expectedOutput = normalizeJUnitOutput(runResult);
+
+    System.out.println("Minimizing: " + filePath);
+
+    // Minimize the Java test suite.
+    minimizeTestSuite(
+        compUnit,
+        packageName,
+        minimizedFile,
+        classPath,
+        expectedOutput,
+        timeoutLimit,
+        verboseOutput);
+
+    // Cleanup: simplify type names and sort the import statements.
+    compUnit =
+        simplifyTypeNames(
+            compUnit, packageName, minimizedFile, classPath, expectedOutput, timeoutLimit);
+
+    writeToFile(compUnit, minimizedFile);
+
+    System.out.println("Minimizing complete.\n");
+    System.out.println("Original file length: " + getFileLength(originalFile) + " lines.");
+    System.out.println("Minimized file length: " + getFileLength(minimizedFile) + " lines.");
+
+    return true;
+  }
+
+  /**
+   * Visit and minimize every JUnit test method within a compilation unit.
+   *
+   * @param cu the compilation unit to minimize; is modified by side effect
+   * @param packageName the package that the Java file is in
+   * @param file the Java file that is being minimized; is modified by side effect
+   * @param classpath classpath used to compile and run the Java file
+   * @param expectedOutput expected JUnit output when the Java file is compiled and run
+   * @param timeoutLimit number of seconds allowed for the whole test suite to run
+   * @param verboseOutput whether to print information about minimization status
+   */
+  private static void minimizeTestSuite(
+      CompilationUnit cu,
+      String packageName,
+      File file,
+      String classpath,
+      Map<String, String> expectedOutput,
+      int timeoutLimit,
+      boolean verboseOutput) {
+    for (TypeDeclaration type : cu.getTypes()) {
+      for (BodyDeclaration member : type.getMembers()) {
+        if (member instanceof MethodDeclaration) {
```

Comment:
Would it be reasonable to attempt to simplify variable names in the next version of the minimizer?